src/ome_zarr_models/v04/multiscales.py

```diff
@@ -277,7 +285,63 @@ class MultiscaleGroupAttrs(Base):
     omero: Omero | None = None
 
 
+def _check_datasets_exist(data: MultiscaleGroup) -> MultiscaleGroup:
+    """
+    Check that the datasets referenced in the `multiscales` metadata are actually contained in this group.
+    """
+    attrs = data.attributes
+    flattened = data.to_flat()
+
+    for multiscale in attrs.multiscales:
+        for dataset in multiscale.datasets:
+            dpath = "/" + dataset.path.lstrip("/")
+            if dpath in flattened:
+                if not isinstance(flattened[dpath], ArraySpec):
+                    msg = (
+                        f"The node at {dpath} should be an array, "
+                        f"found {type(flattened[dpath])} instead"
+                    )
+                    raise ValueError(msg)
```

Comment:
Do we want to eagerly error on the first error we find, or collect up a bunch of errors and then raise a message at the end that includes a message with all the errors found? I'd be inclined to collect them up, so the user can fix them all in one go, instead of having to fix > run code > fix > run code... to find and fix them all.

Reply:
so it's not easy to gather all of the exceptions here, for reasons that I can expound on if needed. I think people might have to live with incomplete "your data is broken" messages unless pydantic changes things about the execution order of validation functions.